crates/bevy_window/src/window.rs

```diff
@@ -127,6 +137,8 @@ pub struct Window {
     mode: WindowMode,
     #[cfg(target_arch = "wasm32")]
     pub canvas: Option<String>,
+    icon_path: Option<PathBuf>,
+    icon: Option<WindowIcon>,
```

Comment:
I like how we now have the ability to either use a path or raw bytes :)

However storing both directly in the window (and exposing functions) doesn't really make much sense since what does it mean for both of them to be set? Which one has precedence?

I would recommend coalescing them into an enum:
```rust
struct WindowIconBytes {
    bytes: Vec<u8>,
    // Width, height, etc...
}

enum WindowIcon {
    Path(PathBuf),
    Bytes(WindowIconBytes),
}
```
then you can implement `From<T> where T: AsRef<Path>` and `From<WindowIconBytes>` for the enum type :)

Reply:
Yeah, I did not like the way this would yield inconsistent data to the outside based on how the icon was set. 
Not sure why I did not think of this myself :D